### PR TITLE
Chapter 9

### PR DESCRIPTION
**Pull request by hoangitsolutions**
screen shot 2018-04-19 at 1 34 15 pm

### CODE REVIEW

**Comment by hoangitsolutions** [comment]
ready

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -2,18 +2,23 @@ class SessionsController < ApplicationController
   def new; end
 
   def create
-    user = User.find_by(email: params[:session][:email].downcase)
-    if user && user.authenticate(params[:session][:password])
+    user = User.find_by email: params[:session][:email].downcase
+    if user && password_authenticate?(user)
       log_in user
+      params[:session][:remember_me] == "1" ? remember(user) : forget(user)
```
Comment (hhhhuy112) [inline app/controllers/sessions_controller.rb]: "1" cho vào Settings.yml đi em

File: app/models/user.rb [hunk 2]
```diff
@@ -1,9 +1,35 @@
...
+  end
+
+  def remember
+    self.remember_token = User.new_token
+    self.remember_digest = User.digest remember_token
+    save(validate: false)
+  end
+
+  def authenticated? remember_token
+    return false if remember_digest.nil?
```
Comment (hhhhuy112) [inline app/models/user.rb]: blank? thay nil?

**Comment by hhhhuy112** [comment]
commented

**Comment by hoangitsolutions** [comment]
ready

**Review by hhhhuy112 — APPROVED** [review]

**Comment by hhhhuy112** [comment]
conflicted
conflicted

**Comment by hoangitsolutions** [comment]
ready

**Review by hhhhuy112 — APPROVED** [review]

**Comment by hhhhuy112** [comment]
merged
Mợt

**Pull request merged**